src_c/rwobject.c

```diff
@@ -437,21 +437,10 @@ pgRWopsFromFileObject(PyObject *obj)
 static int
 pgRWopsReleaseObject(SDL_RWops *context)
 {
-    if (pgRWopsCheckObject(context)) {
-        pgRWHelper *helper = (pgRWHelper *)context->hidden.unknown.data1;
-        Py_XDECREF(helper->seek);
-        Py_XDECREF(helper->tell);
-        Py_XDECREF(helper->write);
-        Py_XDECREF(helper->read);
-        Py_XDECREF(helper->close);
-        PyMem_Del(helper);
-        SDL_FreeRW(context);
-    } else {
-        int ret;
-        if ((ret = SDL_RWclose(context)) < 0) {
-            RAISE(PyExc_IOError, SDL_GetError());
-            return ret;
-        }
+    int ret;
+    if ((ret = SDL_RWclose(context)) < 0) {
```

Comment:
Forcing the font file closed will make it unusable for others that are relying on it being open. It is probably best to just close the file where it has been opened. With this change the following code will just show a blank screen.

```
import time
import pygame
import pygame.freetype

pygame.init()

screen = pygame.display.set_mode((200, 100))
font_file_obj = open(r'c:\windows\fonts\arial.ttf', 'rb')

ft1 = pygame.freetype.Font(font_file_obj, 20)
ft2 = pygame.freetype.Font(font_file_obj, 40)
del ft1
ft2_text, _ = ft2.render('test font', pygame.Color('white'),
                         pygame.Color('blue'))

screen.blit(ft2_text, (0, 0))
pygame.display.flip()
time.sleep(2)

pygame.quit()
```

Reply:
Agreed. I'll make it close the file only if the refcount is 1. Then it should only warn if the user fails to close the handle. If pygame has the only reference to the handle, then it should be safe to close.